minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/test/java/org/apache/nifi/minifi/c2/command/FlowPropertyEncryptorTest.java

```diff
@@ -0,0 +1,338 @@
+
+@ExtendWith(MockitoExtension.class)
+public class FlowPropertyEncryptorTest {
+
+    private static final String PROCESSOR_TYPE_1 = "processor_type_1";
+    private static final String PROCESSOR_TYPE_2 = "processor_type_2";
+    private static final String PROCESSOR_TYPE_3 = "processor_type_3";
+    private static final String CONTROLLER_SERVICE_TYPE_1 = "controller_service_type_1";
+    private static final String CONTROLLER_SERVICE_TYPE_2 = "controller_service_type_2";
+    private static final String CONTROLLER_SERVICE_TYPE_3 = "controller_service_type_3";
+
+    private static final String SENSITIVE_PROPERTY_NAME_PREFIX = "sensitive";
+
+    private static final String NON_SENSITIVE_1 = "non-sensitive-1";
+    private static final String SENSITIVE_1 = SENSITIVE_PROPERTY_NAME_PREFIX + "-1";
+    private static final String NON_SENSITIVE_2 = "non-sensitive-2";
+    private static final String SENSITIVE_3 = SENSITIVE_PROPERTY_NAME_PREFIX + "-3";
+
+    private static final Map<String, String> PARAMETERS1 = Map.of(
+        NON_SENSITIVE_1, NON_SENSITIVE_1,
+        SENSITIVE_1, SENSITIVE_1
+    );
+    private static final Map<String, String> PARAMETERS2 = Map.of(
+        NON_SENSITIVE_2, NON_SENSITIVE_2
+    );
+
+    private static final Map<String, String> PARAMETERS3 = Map.of(
+        SENSITIVE_3, SENSITIVE_3
+    );
+    private static final Map<String, VersionedPropertyDescriptor> DESCRIPTORS1 = Map.of(
+        NON_SENSITIVE_1, versionedPropertyDescriptor(NON_SENSITIVE_1, false),
+        SENSITIVE_1, versionedPropertyDescriptor(SENSITIVE_1, true)
+    );
+    private static final Map<String, VersionedPropertyDescriptor> DESCRIPTORS2 = Map.of(
+        NON_SENSITIVE_2, versionedPropertyDescriptor(NON_SENSITIVE_2, false)
+    );
+    private static final Map<String, VersionedPropertyDescriptor> DESCRIPTORS3 = Map.of(
+        SENSITIVE_3, versionedPropertyDescriptor(SENSITIVE_3, true)
+    );
+
+
+    @Mock
+    private PropertyEncryptor mockPropertyEncryptor;
+    @Mock
+    private RuntimeManifest mockRunTimeManifest;
+
+    private FlowSerDeService flowSerDeService;
+    private FlowPropertyEncryptor testEncryptor;
+
+    private static VersionedPropertyDescriptor versionedPropertyDescriptor(String name, boolean isSensitive) {
+        VersionedPropertyDescriptor versionedPropertyDescriptor = new VersionedPropertyDescriptor();
+        versionedPropertyDescriptor.setName(name);
+        versionedPropertyDescriptor.setSensitive(isSensitive);
+        return versionedPropertyDescriptor;
+    }
+
+    @BeforeEach
+    public void setup() {
+        when(mockPropertyEncryptor.encrypt(anyString())).thenReturn(randomAlphabetic(5));
+        flowSerDeService = FlowSerDeService.defaultInstance();
+        testEncryptor = new FlowPropertyEncryptor(mockPropertyEncryptor, flowSerDeService, mockRunTimeManifest);
+    }
+
+    @Test
+    public void shouldEncryptParameterContextsSensitiveVariables() {
+        // given
```

Comment:
The `given/when/then` comments should be removed as they are not standard across the repository and do not necessarily add much to the test definition.

Reply:
Removed the comments